runelite-client/src/main/java/net/runelite/client/plugins/timers/TimersPlugin.java

```diff
@@ -505,18 +553,64 @@ public void onGameTick(GameTick event)
 		Player player = client.getLocalPlayer();
 		WorldPoint currentWorldPoint = player.getWorldLocation();
 
-		if (freezeTimer == null)
+		if (freezeTimer != null)
+		{
+			// assume movement means unfrozen
+			if (!currentWorldPoint.equals(lastPoint))
+			{
+				removeGameTimer(freezeTimer.getTimer());
+				freezeTimer = null;
+			}
+		}
+
+		lastPoint = currentWorldPoint;
+
+		if (!gameStateLoggedIn
+			&& !widgetHiddenChangedOnPvpWorld)
 		{
-			lastPoint = currentWorldPoint;
 			return;
 		}
 
-		// assume movement means unfrozen
-		if (!currentWorldPoint.equals(lastPoint))
+		if (gameStateLoggedIn)
+		{
+			gameStateLoggedIn = false;
+
+			PvpWorldType worldType = worldSetToType(client.getWorldType());
+			if (lastWorldType != worldType)
+			{
+				if (worldType == DMM || lastWorldType == DMM
+					|| worldType == SDMM || lastWorldType == SDMM)
+				{
+					log.debug("Hopped to or from a DMM world, clearing Teleblock timer.");
```

Comment:
This also probably needs to account for TOURNAMENT worlds

Reply:
Tournament worlds are now included in the "different account type" worlds.